ibis/expr/timecontext.py

```diff
@@ -127,6 +127,88 @@ def canonicalize_context(
     return begin, end
 
 
+def construct_time_context_aware_series(
+    series: pd.Series, frame: pd.DataFrame
+) -> pd.Series:
+    """ Construct a Series by adding 'time' in its MultiIndex
+
+    In window execution, the result Series of udf may need
+    to be trimmed by timecontext. In order to do so, 'time'
+    must be added as an index to the Series. We extract
+    time column from the parent Dataframe `frame`.
+    See `trim_with_timecontext` in execution/window.py for
+    trimming implementation.
+
+    Examples:
+        Assume frame is:
+                             time  id  value
+        0 2017-01-02 01:02:03.234   1    1.1
```

Comment:
can you actually programatically construct this, *then* show it.

also use
```
Examples
-----------
```

Reply:
IOW do this in ipython and basically copy it here (or alternatly do it in a python terminal and use `>>>`

Reply:
idea is that this code can be copy pasted to a terminal for execution directly; so include imports too.

Reply:
Good point thx